cluster-autoscaler/processors/customresources/dra_processor.go

```diff
@@ -57,11 +58,21 @@ func (p *DraCustomResourcesProcessor) FilterOutNodesWithUnreadyResources(autosca
 			continue
 		}
 
-		nodeInfo, err := ng.TemplateNodeInfo()
-		if err != nil {
-			newReadyNodes = append(newReadyNodes, node)
-			klog.Warningf("Failed to get template node info for node group %s with error: %v", ng.Id(), err)
-			continue
+		var nodeInfo *framework.NodeInfo
+		if autoscalingCtx.TemplateNodeInfoRegistry != nil {
```

Comment:
I think we can rely on `autoscalingCtx.TemplateNodeInfoRegistry` not being nil if we always set it during `NewStaticAutoscaler`. Having the nil check suggests to the reader that this field can be nil, but that should never happen, right?